lib/api/entities.rb

```diff
@@ -251,9 +251,18 @@ class ProjectService < Grape::Entity
       expose :push_events, :issues_events, :merge_requests_events, :tag_push_events, :note_events
       # Expose serialized properties
       expose :properties do |service, options|
-        field_names = service.fields.
-          select { |field| options[:include_passwords] || field[:type] != 'password' }.
-          map { |field| field[:name] }
+        field_names = []
+        service.fields.each do |field|
+          if field[:type] == 'fieldset'
+            field[:fields].each do |subfield|
+              if options[:include_passwords] || subfield[:type] != 'password'
```

Comment:
In addition, subfields should not iterate recursively. 
I've come to think that this function is way more concise and maintainable *as is* with this duplication within the span of three lines compared to any other alternative involving a helper method, recursion, nesting level etc.